srcpkgs/boost/patches/align.patch

```diff
@@ -0,0 +1,49 @@
+This code is wrong and uses feature test macros to determine whether something
+is available; FTMs shouldn't be in user code outside of top level #define's.
+
+This lead to an issue on musl where libtorrent-rasterbar was including this
+header (via boost asio) from two different places, and segfaulting due to
+mismatched implementations -- one's the posix one using posix_memalign, the
+other one's the generic one that stores a pointer to itself.
+
+posix_memalign is always available and should always be used, so we force that.
+
+diff --git a/boost/align/aligned_alloc.hpp b/boost/align/aligned_alloc.hpp
+index 1d81a13..a441645 100644
+--- a/boost/align/aligned_alloc.hpp
++++ b/boost/align/aligned_alloc.hpp
+@@ -14,34 +14,6 @@ Distributed under the Boost Software License, Version 1.0.
+ #include <unistd.h>
+ #endif
+ 
+-#if defined(__APPLE__) || defined(__APPLE_CC__) || defined(macintosh)
```

Comment:
Or `#error` in the last leg

Reply:
I don't want to fix this with a define in CXXFLAGS because it requires fixing in all packages that use the library, instead of a single fix in the library.

If I manage to rework the patch to look like the posix version is the one being kept or add a comment making that clear and that it's because, as a distro, we know where these headers are being used, would it be okay for you?

Reply:
I guess reworking the last fallback to just include the posix version would be equivalent, and I'd be okay with it.

Reply:
I think it's better to `#include <features.h>` in `boost/config.hpp`. Just in case other headers (now or later) decide to use feature test macros.